packages/calypso-e2e/src/lib/pages/user-signup-page.ts

```diff
@@ -63,10 +69,18 @@ export class UserSignupPage {
 		// user to this page.
 		// Example: invite => Signup & Follow
 		//          signup from login => Create your account
-		await Promise.all( [
+		const [ , response ] = await Promise.all( [
 			this.page.waitForNavigation(),
+			this.page.waitForResponse( /.*new\?.*/ ),
 			this.page.click( selectors.submitButton ),
 		] );
+
+		if ( ! response ) {
+			throw new Error( 'Failed to create new user at signup.' );
+		}
+
+		const responseBody: NewUserResponse = JSON.parse( ( await response.body() ).toString() );
+		return { ID: responseBody.body.user_id, bearer_token: responseBody.body.bearer_token };
```

Comment:
This is similar to what we do in `EditorPage.publish`. By intercepting and decoding the response when we create a new user (via signup), the data can then be used to cross-reference that we are operating on the expected user when interacting with the API.